src/lib.rs

```diff
@@ -268,6 +268,55 @@ impl<T: AsBytes, X> LocatedSpan<T, X> {
         &self.fragment
     }
 
+    /// Transform the extra inside into another type
+    ///
+    /// # Example of use
+    /// ```
+    /// # extern crate nom_locate;
+    /// # extern crate nom;
+    /// # use nom_locate::LocatedSpan;
+    /// use nom::{
+    ///   IResult,
+    ///   multi::{many0, many1},
+    ///   combinator::{recognize, map_res},
+    ///   sequence::{terminated, tuple},
+    ///   character::complete::{char, one_of},
+    ///   bytes::complete::tag
+    /// };
+    ///
+    /// fn decimal(input: LocatedSpan<&str>) -> IResult<LocatedSpan<&str>, LocatedSpan<&str>> {
+    ///   recognize(
+    ///     many1(
+    ///       terminated(one_of("0123456789"), many0(char('_')))
+    ///     )
+    ///   )(input)
+    /// }
+    ///
+    /// fn main() {
+    ///     let span = LocatedSpan::new("$10");
+    ///     // matches the $ and then matches the decimal number afterwards,
+    ///     // converting it into a `u8` and putting that value in the span
+    ///     let (_, (_, n)) = tuple((
+    ///                         tag("$"),
+    ///                         map_res(
+    ///                             decimal,
+    ///                             |x| x.fragment().parse::<u8>().map(|n| x.map(|_| n))
+    ///                         )
+    ///                       ))(span).unwrap();
+    ///     assert_eq!(n.extra, 10);
+    /// }
+    /// ```
+    pub fn map<U, F: Fn(X) -> U>(self, f: F) -> LocatedSpan<T, U> {
```

Comment:
`FnOnce` should work and be more generic.

Would it make sense to rename this to `map_extra`, just so it is unambiguous what is being mapped? (ie. not the fragment)

Reply:
Yeah, right, tomorrow if I have time I'll apply that

Reply:
Done, and the doctest is fixed now